### ISSUE TITLE
Producing self-contained HTML?

### ISSUE THREAD

**Issue by FlorianMarquardt** [lsarra/linkmap#4]
One original design choice was to keep everything inside one HTML. The goal was that eventually there would be a button that allows the user to download the entire linkmap (including javascript logic) as a self-contained HTML that can then be treated like any normal file (stored in any folder, sent via email etc.) and is fully stand-alone, without any dependencies.

Is it still possible to implement that back?

Such an option will certainly accelerate the adoption of linkmap by the community, if you do not need to set up a dedicated server for the purpose of using it.

**Comment by lsarra** [comment]
That's a design choice. It would be possible, but it decreases the readability of the code. Above all, every map would end up carrying different versions of the code - so something may work in a file but not in another. Up to now the third button at the top would provide the download of only the code of the linkmap,

What I would propose is to give the possibility to download a linkmap html file, which would include the entire map and it would automatically download the required (and updated) dependencies (e.g. from this repository). 

If we really want to embed css and js in the stand-alone file, I would do it with some javascript in the file that the user downloads, not in the source of the page.

**Comment by FlorianMarquardt** [comment]
I agree the option to download a stand-alone html should be only an additional option, not the default.

But, the whole purpose was to avoid dependencies on other web-sites or even other files, so linking will not really help that much.

A stand-alone html page will never break in the future, that is the great advantage.

**Comment by lsarra** [comment]
OK, one could easily implement this. The problem is what should happen when one edits one of such stand-alone notes. The browser is not allowed to directly write the changes to disk. This implies that one should always manually save the file again after some changes (or the changes would be lost). 

An alternative could be to make a stand-alone app with Electron and set it as the default reader of files with extension .linkmap.
